src/dune_config_file/dune_config_file.ml

```diff
@@ -16,6 +16,35 @@ module Dune_config = struct
      simplicity *)
   let syntax = Stanza.syntax
 
+  module Project_defaults = struct
+    type t =
+      { authors : string list option
```

Comment:
Can't this be `string list`? If you aren't allowing the empty list, then `[]` can just mean the field is left unspecified. `string list option` is only useful if `(authors)` is allowed.

Reply:
It could be changed to `string list`. I think my original intention was to just copy the allowed behavior in the `dune-project` file, when I was testing it seems that `(authors)` is allowed there and didn't cause any issue.

Also in [src/dune_lang/package_info.mli](https://github.com/ocaml/dune/pull/10835/files) the function `example` takes a `string list option` where a value of `[]` literally means there were no arguments provided and a value of `None` uses the default value of `Author Name <author@example.com>`.

Let me know what you think, and if needed I can change it!

Reply:
I see what you mean by just having the empty list represent the absence of the stanza, I also figured allowing something like `(authors)` was a bit less error prone for a user especially when it came to throwing errors when someone is trying to make a new project.

Reply:
In `Package_info.decode` don't we allow the empty list for these fields? In that case, `string list option` indeed makes sense there. My objection is that your code pretends that `None` and `Some []` are possible, but validation in fact forbids that.

Reply:
In `package_info.ml` authors, license, and maintainers are all decoded as a `string list option`. The encoder also uses the same scheme defaulting to the empty list if a value of `None` is provided.

Reply:
Also to further clarify, each of the sub stanzas of `project_defaults` is optional so you could do something like this
```
(project_defaults
 (authors)
 (license MIT))
```
in this case authors would have the empty list value and maintainers would have a value of `None`

Reply:
> is optional so you could do something like this

Looking at your code, I don't think your example will work. You're using `repeat1` which will forbid `(authors)`.

Reply:
I guess your right! What way should we have it then? Should we allow `(authors)` or should I simplify it down to just use `string list` over `string list option`

Reply:
Both are fine I think. On one hand, we allow the empty file in the `dune-project` file. So it would be more consistent to allow it here as well. On the other hand, the empty list is fairly useless as a default in this case.

Reply:
I think we should allow it just to keep constancy, that way you can edit your `config` file as if you were editing your `dune-project` file!